### PR TITLE
fix(testing): preserve non-enumerable properties of pinia instance in createTestingPinia

### LINKED ISSUE #827: Error with `createTestingPinia` in vue 2
### Reproduction

https://stackblitz.com/github/Djaler/pinia-vue-2-test?file=src%2F__tests__%2FApp.spec.js

### Steps to reproduce the behavior

1. Open stackblitz link
2. Wait for dependencies install
3. Run `npx vue-demi-switch 2` (looks like stackblitz doesn't autorun postinstall)
4. Run `npm run test`

### Expected behavior

No errors

### Actual behavior

```
Cannot read properties of undefined (reading '__ob__')
TypeError: Cannot read properties of undefined (reading '__ob__')
    at Object.set$1 [as set] (D:\Projects\pinia-vue-2-test\node_modules\@vue\composition-api\dist\vue-composition-api.common.js:1001:21)
    at setup (D:\Projects\pinia-vue-2-test\node_modules\pinia\dist\pinia.cjs:1117:25)
    at D:\Projects\pinia-vue-2-test\node_modules\pinia\dist\pinia.cjs:1356:32
    at EffectScope.Object.<anonymous>.EffectScopeImpl.run (D:\Projects\pinia-vue-2-test\node_modules\@vue\composition-api\dist\vue-composition-api.common.js:201:24)
    at D:\Projects\pinia-vue-2-test\node_modules\pinia\dist\pinia.cjs:1356:22
    at EffectScope.Object.<anonymous>.EffectScopeImpl.run (D:\Projects\pinia-vue-2-test\node_modules\@vue\composition-api\dist\vue-composition-api.common.js:201:24)
    at createSetupStore (D:\Projects\pinia-vue-2-test\node_modules\pinia\dist\pinia.cjs:1354:33)
    at createOptionsStore (D:\Projects\pinia-vue-2-test\node_modules\pinia\dist\pinia.cjs:1145:13)
    at useStore (D:\Projects\pinia-vue-2-test\node_modules\pinia\dist\pinia.cjs:1621:17)
    at setup (D:\Projects\pinia-vue-2-test\src\App.vue:16:1)
    at mergedSetupFn (D:\Projects\pinia-vue-2-test\node_modules\@vue\composition-api\dist\vue-composition-api.common.js:2099:113)
    at mergedSetupFn (D:\Projects\pinia-vue-2-test\node_modules\@vue\composition-api\dist\vue-composition-api.common.js:2099:51)
    at D:\Projects\pinia-vue-2-test\node_modules\@vue\composition-api\dist\vue-composition-api.common.js:1915:23
    at activateCurrentInstance (D:\Projects\pinia-vue-2-test\node_modules\@vue\composition-api\dist\vue-composition-api.common.js:1830:16)
    at initSetup (D:\Projects\pinia-vue-2-test\node_modules\@vue\composition-api\dist\vue-composition-api.common.js:1913:9)
    at VueComponent.wrappedData (D:\Projects\pinia-vue-2-test\node_modules\@vue\composition-api\dist\vue-composition-api.common.js:1896:13)
    at getData (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:4745:17)
    at initData (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:4702:7)
    at initState (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:4641:5)
    at VueComponent.Vue._init (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:5001:5)
    at new VueComponent (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:5148:12)
    at createComponentInstanceForVnode (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:3298:10)
    at init (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:3127:45)
    at createComponent (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:6002:9)
    at createElm (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:5949:9)
    at VueComponent.patch [as __patch__] (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:6499:7)
    at VueComponent.Vue._update (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:3948:19)
    at VueComponent.updateComponent (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:4069:10)
    at Watcher.get (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:4481:25)
    at new Watcher (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:4470:12)
    at mountComponent (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:4076:3)
    at VueComponent.Object.<anonymous>.Vue.$mount (D:\Projects\pinia-vue-2-test\node_modules\vue\dist\vue.runtime.common.dev.js:8436:10)
    at mount (D:\Projects\pinia-vue-2-test\node_modules\@vue\test-utils\dist\vue-test-utils.js:14057:21)
    at Object.<anonymous> (D:\Projects\pinia-vue-2-test\src\__tests__\App.spec.js:10:21)
```

### PR DESCRIPTION
**Pull request by Djaler**
Closes #827

### CODE REVIEW

**Comment by Djaler** [comment]
Sorry, I don't know how to cover this by tests, because this problem exists only when using with Vue 2 and `@vue/composition-api`, but current tests are all for Vue 3

File: packages/testing/src/testing.ts [hunk 1]
```diff
@@ -93,12 +93,11 @@ export function createTestingPinia({
 
   setActivePinia(pinia)
 
-  return Object.assign(
-    {
-      get app(): App {
-        return (this as TestingPinia)._a
-      },
+  Object.defineProperty(pinia, 'app', {
+    get: function (): App {
```
Comment (posva) [inline packages/testing/src/testing.ts]: Let's keep it like before:

```suggestion
    configurable: true,
    enumerable: true,
    get(): App {
```

File: packages/testing/src/testing.ts [hunk 2]
```diff
@@ -93,12 +93,11 @@ export function createTestingPinia({
...
+      return (this as TestingPinia)._a
```
Comment (posva) [inline packages/testing/src/testing.ts]: I think the `as ...` can be removed now

**Review by posva — CHANGES_REQUESTED** [review]
Thanks, I will have to test this manually as it's quite difficult to have tests for Vue 3 and Vue 2 in the same repo

**Comment by Djaler** [comment]
> as it's quite difficult to have tests for Vue 3 and Vue 2 in the same repo

Maybe we can encapsulate setup code into functions that will work for both vue 2 and vue 3, like
```
function mountComponentWithPinia(component) {
    if (isVue2) {
      const localVue = createLocalVue();
      const pinia = createTestingPinia();
      return mount(component, {
        localVue,
        pinia,
      })
    } else {
      return mount(component, {
        global: {
          plugins: [createTestingPinia()],
        },
      })
    }
  }
  ```
  and then do something like
  ```
  "scripts": {
    "test:vue2": "vue-demi-switch 2 && jest",
    "test:vue3": "vue-demi-switch 3 && jest",
  }
  ```

**Review by posva — APPROVED** [review]

**Comment by posva** [comment]
Last time I tried, that didn't work, things were blocking somewhere else 🤔

**Pull request merged**